### PR TITLE
[CI][Clean][4] Add exception-free functions

### PR DESCRIPTION
**Pull request by can-anyscale**
## Why are these changes needed?

Add exception-free APIs for some classes. This helps client with the option to use them without having to worry about exception handling repetitively. Make the client code a bit easier to read.

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
   - [X] Release tests

### CODE REVIEW

**Comment by can-anyscale** [comment]
ping ping @aslonnie @krfricke @scv119

**Review by aslonnie — APPROVED** [review]
change looks good overall. I am not very sure about the background/motivation though.

**Comment by aslonnie** [comment]
> not very sure about the background/motivation though.

like in general, I thought that errors should fail loudly. but maybe these error that we are only logging are not real errors?

File: release/ray_release/command_runner/command_runner.py [hunk 1]
```diff
@@ -118,10 +119,17 @@ def run_prepare_command(
...
+            return self.get_last_logs_ex()
+        except Exception as e:
+            logger.exception(f"Error fetching logs: {e}")
+            return "No logs could be retrieved."
+
+    def get_last_logs_ex(self):
         raise NotImplementedError
 
-    def fetch_results(self) -> Dict[str, Any]:
+    def fetch_results_ex(self) -> Dict[str, Any]:
```
[the remark this answers is not in the document]
Comment (can-anyscale) [reply release/ray_release/command_runner/command_runner.py]: yeah that's what I intended ;)

File: release/ray_release/glue.py [hunk 2]
```diff
@@ -540,20 +540,11 @@ def run_release_test(
...
-        last_logs = command_runner.get_last_logs()
-    except Exception as e:
-        logger.exception(f"Error fetching logs: {e}")
-        last_logs = "No logs could be retrieved."
+    result.last_logs = command_runner.get_last_logs()
 
-    result.last_logs = last_logs
-
-    if not no_terminate and isinstance(cluster_manager, FullClusterManager):
+    if not no_terminate:
```
[the remark this answers is not in the document]
Comment (can-anyscale) [reply release/ray_release/glue.py:545]: Require more brain energy for sure ;). It's a flag to ask anyscale to keep the cluster alive after it finishes the job

[the remark this answers is not in the document]
Comment (aslonnie) [reply release/ray_release/glue.py:545]: ok :) maybe rename to `keep_alive` or something?

[the remark this answers is not in the document]
Comment (can-anyscale) [reply release/ray_release/glue.py:545]: I like that better ;). I'll do that in a follow up PR, that names come from a command-line flag so needs a bit more refactoring

**Comment by can-anyscale** [comment]
I think this glue file is already close to the user level so they want to handle all errors nicely

**Review by scv119 — APPROVED** [review]

**Pull request merged**